http/controllers/UserController.ts

```diff
@@ -15,30 +16,74 @@ export class UserController {
     const userId: number = parseInt(ctx.params.id)
     const user: User = this.users.find(user => user.id === userId)
 
+    if (!user) {
+      ctx.status = 404
+      ctx.body = {
+        error: 'User does not exist',
+      }
+
+      return
+    }
+
     ctx.body = {
-      user,
+      user: {
+        firstName: user.firstName,
+        lastName: user.lastName,
+        username: user.username,
+        email: user.email,
+      },
     }
   }
 
   create = async (ctx: IRouterContext) => {
     const user: User = ctx.request.body
-    user.id = this.users.length + 1
+    const emailExists: boolean = this.users.some(u => u.email === user.email)
+    const usernameExists: boolean = this.users.some(
+      u => u.username === user.username
+    )
+    const saltRounds: number = 10
+
+    const hash = await bcrypt.hash(user.password, saltRounds)
```

Comment:
should do this after the check for username and email exists, since this is an expensive function to run, and there is a possibility you will just return without using it